Production_scripts/Process_extracts/extract_delayed_discharges.R

```diff
@@ -0,0 +1,94 @@
+#####################################################
+# Draft pre processing code for Delayed Discharges
+# Author: Jennifer Thom
+# Date: April 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input - Jul16_{latest dd period}DD_LinkageFile.zssav
+# Description - Preprocessing of raw delayed discharges file.
+#               Tidy up file in line with SLF format
+#                prior to processing.
+#####################################################
+
+# Load Packages #
+library(dplyr)
+library(janitor)
+library(lubridate)
+library(createslf)
+
+
+# Read in data---------------------------------------
+
+year <- check_year_format("1920")
+
+dd_file <- haven::read_sav(get_dd_path(ext = "zsav")) %>%
+  clean_names() %>%
+  # rename variables
+  rename(
+    keydate1_dateformat = rdd,
+    keydate2_dateformat = delay_end_date
+  )
+
+
+# Data Cleaning---------------------------------------
+
+dd_clean <- dd_file %>%
+  # Use end of the month date for records with no end date (but we think have ended)
+  # Create a flag for these records
+  mutate(
+    month_end = lubridate::ceiling_date(keydate1_dateformat, "month") - 1,
+    ammended_dates = case_when(
+      keydate2_dateformat == as.Date("1900-01-01") ~ TRUE,
+      TRUE ~ FALSE
+    ),
+    keydate2_dateformat = if_else(ammended_dates == TRUE, month_end, keydate2_dateformat)
+  ) %>%
+  # Drop any records with obviously bad dates
+  filter(
+    (keydate1_dateformat <= keydate2_dateformat) | is.na(keydate2_dateformat)
+  ) %>%
+  # set up variables
+  mutate(
+    recid = "DD",
+    smrtype = "DelayedDis",
+    year = year
+  ) %>%
+  # recode blanks to NA
+  mutate(
+    mutate(across(ends_with("delay_reason"), na_if, ""))
+  ) %>%
+  # create flags for no_end_date and correct_dates
+  create_dd_flags(year) %>%
```

Comment:
I don't really like this being extracted as a function, as the function is only used in this one place and is very specific I think it just makes the code less readable.

Could you just put the code from the function `create_dd_flags` back in-line, please?